Dockerfile

```diff
@@ -30,11 +32,31 @@ RUN DEBIAN_FRONTEND=noninteractive && apt-get update && apt-get dist-upgrade -y
   vim \
   && rm -rf /var/lib/apt/lists/*
 
-WORKDIR /src
+ENV TINI_VERSION v0.18.0
+COPY ./tini-install.sh /tini-install.sh
+RUN chmod +x /tini-install.sh && TARGETPLATFORM=linux/amd64 /tini-install.sh && rm /tini-install.sh
+
+RUN mkdir /etc/inboxapp && \
+  chown sync-engine:sync-engine /etc/inboxapp && \
+  mkdir /var/lib/inboxapp && \
+  chown sync-engine:sync-engine /var/lib/inboxapp && \
+  mkdir /opt/app && \
+  chown sync-engine:sync-engine /opt/app && \
+  mkdir /opt/venv && \
+  chown sync-engine:sync-engine /opt/venv
+
+USER sync-engine
+
+WORKDIR /opt/app
+
+ENV PATH="/opt/venv/bin:$PATH"
+
 COPY ./ ./
-RUN pip install setuptools==44.0.0
-RUN pip install pip==20.3.4
-RUN pip install -r requirements_frozen.txt -e .
+RUN \
+  virtualenv /opt/venv && \
```

Comment:
why use a virtualenv inside a docker image?

Reply:
This was a suggestion from @jkemp101